### PR TITLE
add convecting taylor vortex regression test using edges

### PR DESCRIPTION
**Pull request by ashesh2512**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [ ] Feature enhancement

Add regression test for convecting taylor vortex using edges.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [ ] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [x] Linux
    - [x] MacOS
  - Compilers 
    - [x] GCC
    - [x] LLVM/Clang
    - [ ] Intel compilers
    - [x] NVIDIA CUDA
- [x] Compiles without warnings
- [ ] Passes all unit tests
- [x] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [x] New regression tests exercising the new code

### CODE REVIEW

**Comment by sayerhs** [comment]
Thanks @ashesh2512. It looks good overall. I had a couple of questions:

- Since you're outputting solution norms, can implement a check based on solution norms instead of `mean system norms`? That would be preferable because we can set a tighter tolerance for all compilers.
- Would it be possible to run this test on GPUs also? If we do that we should tighten the linear solver tolerances.

**Comment by ashesh2512** [comment]
@sayerhs I have pushed an additional commit to modify the test to check against solution norms - `845196205b4808870f7938e55014b030bd11fc9d`

File: src/SolutionNormPostProcessing.C [hunk 1]
```diff
@@ -339,6 +339,12 @@ SolutionNormPostProcessing::execute()
     int offSet = 0;
     for ( size_t j = 0; j < fieldPairVec_.size(); ++j ) {
 
+      // sync fields
+      fieldPairVec_[j].first->modify_on_device();
```
Comment (sayerhs) [inline src/SolutionNormPostProcessing.C:343]: @ashesh2512 These should be marked as modified where they were updated. Only need `sync_to_host` here.

File: src/SolutionNormPostProcessing.C [hunk 2]
```diff
@@ -339,6 +339,12 @@ SolutionNormPostProcessing::execute()
...
+      fieldPairVec_[j].first->sync_to_host();
+      fieldPairVec_[j].second->modify_on_device();
+      fieldPairVec_[j].second->sync_to_host();
```
Comment (sayerhs) [inline src/SolutionNormPostProcessing.C:346]: Exact DOF field is `user_functions` and don't have NGP implementations. I suppose we are computing exact fields again somewhere after this?

File: src/SolutionNormPostProcessing.C (hunk 1 above)
Comment (ashesh2512) [reply src/SolutionNormPostProcessing.C]: @sayerhs Thanks for pointing that out. I was trying to be safe, but I have pushed a commit removing the redundant calls - https://github.com/Exawind/nalu-wind/pull/838/commits/4911b48139317aaee25402b59baa3fbb4a35a4e6

File: src/SolutionNormPostProcessing.C (hunk 2 above)
Comment (ashesh2512) [reply src/SolutionNormPostProcessing.C]: @sayerhs Thanks for catching this. You are right, the exact fields should not be modified anywhere on device. I have removed these calls in https://github.com/Exawind/nalu-wind/commit/4911b48139317aaee25402b59baa3fbb4a35a4e6

**Review by sayerhs — APPROVED** [review]

**Pull request merged**